### PR TITLE
Switch to kidfearless/Auto-Exec-Config-Class for ConVars and use the ConVar methodmap

### PR DESCRIPTION
**Pull request by azalty**
Check #115 for context and to know the benefits of this change

Tests:
✔️ Compiles fine
❌ Tested in game

Major changes:
- Switched to kidfearless/Auto-Exec-Config-Class
- Switched to the ConVar or Convar methodmaps, and thus, the new syntax
- Fixed 2 conflicts with convar_class by changing the name of 2 global variables. Seems like LilAC likes to register global variables with frequent names! (me angry)

The 2 global variables that got changed are:
`char line[2048]` to `char line_buffer[2048]`: [difference](https://github.com/J-Tanzanite/Little-Anti-Cheat/commit/2be3d1455aada6a600f126d492bb223396b77f51#diff-1994aeabac931852c3d2f269d6ad400002167a5f644a8f87afaa2b89c9483e6cR165)
`Handle cvar[CVAR_MAX]` to `Convar hcvar[CVAR_MAX]` (`h` meaning `Handle`, because it's an Handle): [difference](https://github.com/J-Tanzanite/Little-Anti-Cheat/commit/2be3d1455aada6a600f126d492bb223396b77f51#diff-1994aeabac931852c3d2f269d6ad400002167a5f644a8f87afaa2b89c9483e6cR144)

**Every file in this repo was modified to match the new variable name. It compiles fine for me.**

Minor changes:
- Changed some Format() to FormatEx() when I saw it and when it was possible
- Improved the database module by updating the database config name through lilac_config.sp

Notes:
- The compiled plugin (.smx) isn't included for security reasons. I know you prefer to compile it yourself 😉
- This might need a bit more testing
- Any previously configured cvar will be kept! You won't lose your config.
- You can't comment the cvar config file anymore, or add non-lilac cvars.

### CODE REVIEW

**Comment by azalty** [comment]
Found a problem with the database system due to the change with how the database config name is loaded, fixing it right now.

**Comment by J-Tanzanite** [comment]
I feel bad not having responded to this earlier, as I feel like you might have done a lot of work for nothing, but I've been busy...
Real life stuff is just eating my time, and the free time I do have - spent on other stuff.

That aside, there are some issues I take with this.
I wanna preface by saying that changing the convar system itself, doesn't bother me.
As Sourcemod by default doesn't update ConVars correctly, and I don't just mean the config file.
So I'd LOVE for a change that fixes those issues, but this solution... I'm unsure about...
Here are three concerns I have:
1. I haven't had much time to look at this code, but this convar code, comes from a project with the incorrect license... It's MIT, and legally, should/must be GPL? I am not sure... How this would affect this project.
Obviously, the MIT license would be invalid, and this project would remain under GPL - However, how would we handle accreditation? Or copyright notices? And would importing such code be upsetting to some people?
2. This convar library uses `enum struct`. Last time I did this, it caused older versions of SM to be unable to compile Lilac. And as it turns out, some people do rely on older SM versions... Although there is something to be said about people using old and outdated versions of stuff - putting that aside - I don't think we ought to make changes that make their servers less protected from cheaters.
3. This is possibly an invalid concern, and I may have misunderstood... But quoting the documentation for this convar system: "If a convar was removed from the plugin then it will be removed from the config, **and as a side effect, any convars not belonging to the plugin will be removed.** " ... This could be a huge issue. You see, There have been some cases where I've told people to insert non-lilac ConVars into the lilac config file (because you have to set a lilac convar before setting some other specific convars for some other stuff to work properly)... It's not just that I've told some people to do this, you HAVE to do this for some other stuff to work... If the config file gets non-lilac convars cleansed - that could break stuff.

I have also never worked with methodmaps, and so I can't tell if they would cause the second issue as well.
With those concerns stated, I won't dismiss this solution, but I think we ought to discuss it.

**Comment by azalty** [comment]
>     1. I haven't had much time to look at this code, but this convar code, comes from a project with the incorrect license... It's MIT, and legally, should/must be GPL? I am not sure... How this would affect this project.
>        Obviously, the MIT license would be invalid, and this project would remain under GPL - However, how would we handle accreditation? Or copyright notices? And would importing such code be upsetting to some people?

No one really cares about licenses. But every SourceMod project is basically forced to be under GNU GPL (v2 or v3). The fact that the author picked another license just shows that they didn't even look at the SourceMod license rules. They're just there to look good to be honest, at least at our size.

>     2. This convar library uses `enum struct`. Last time I did this, it caused older versions of SM to be unable to compile Lilac. And as it turns out, some people do rely on older SM versions... Although there is something to be said about people using old and outdated versions of stuff - putting that aside - I don't think we ought to make changes that make their servers less protected from cheaters.

Why don't they just compile the plugins with the newest compilers? I'm pretty sure you can compile plugins with the new compiler while being able to use your old SM installation on your servers. If it doesn't, well they should really just update to 1.10 stable, because, well, it is stable and backwards compatible.

They don't need to compile lilac if you just provide a compiled version.

> It's not just that I've told some people to do this, you HAVE to do this for some other stuff to work... If the config file gets non-lilac convars cleansed - that could break stuff.

Ah, well, that sucks... Those files shouldn't be used for that though 👀
Wouldn't server.cfg work? -> `sm_cvar your_cvar "value"`

> I have also never worked with methodmaps, and so I can't tell if they would cause the second issue as well. With those concerns stated, I won't dismiss this solution, but I think we ought to discuss it.

Methodmaps can only be compiled with recent stable builds, I'm pretty sure they aren't supported by compilers older than 1.10, which, again, shouldn't be a problem as it is the latest stable compiler version that is supported. You really just need to update the compiler by getting it from the latest stable build and you're good to go.

File: scripting/lilac/lilac_globals.sp [hunk 1]
```diff
@@ -150,14 +150,19 @@ int force_disable_bhop = 0;
 /* Banlength overwrite. */
 int ban_length_overwrite[CHEAT_MAX];
 
+/* Database. */
+Database lil_db;
+char sql_buffer[1500]; /* It's probably bigger than what you need, but better be safe than sorry I guess. */
+char db_name[64]; /* Database config name from cvar[CVAR_DATABASE]. */
```
Comment (azalty) [inline scripting/lilac/lilac_globals.sp:156]: It should actually be "hcvar[CVAR_DATABASE]", not "cvar[CVAR_DATABASE]" 👀

Comment (azalty) [reply scripting/lilac/lilac_globals.sp]: Resolved with commit [bb09814](https://github.com/J-Tanzanite/Little-Anti-Cheat/pull/116/commits/bb09814f64a5b197c084a8bf1ca63339399e6438)

**Pull request merged**